### DISCUSSION TITLE
Renovate + GO generated modules

### DISCUSSION THREAD

**Discussion by ghost** [renovatebot/renovate#23400] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

We have a problem with renovate and gererated modules. We have a module, which is referenced by an import in code but will generated on build time with `make`. So if renovate tries to go throw imports with `go get` it cannot resolve the import. 

There is any solution on that? Can I execute `make build` before renovate fetching dependencies? This problem should have many projects working with module generation.

I couldn't find any preUpgradeTasks

### Relevant debug logs

"Command failed: go get -d -t ./...
                          go: downloading ... // so far, so good
                          go: module gitlab.host.com/teams/my-project/gen/go/api: git ls-remote -q origin in 
                          /go/pkg/mod/cache/vcs/6a51fc24ab532668ca2f408807c8506d8fd182bd461b271722c8371665cd3c15: exit 
                          status 128:
                          remote: The project you were looking for could not be found or you don't have permission to view it.
                          fatal: repository 'https://gitlab.host.com/teams.git/' not found"

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment, marked as answer]
preUpgradeTasks feature request: https://github.com/renovatebot/renovate/issues/8804 (in future please search issues and discussions, not just docs)

**Reply by ghost** [reply]
Thx for reply. But this feature is still in progress. I'm not sure, when I can use it.

**Reply by HonkingGoose** [reply]
Hi @vitkep, please follow these steps to know when you can use the feature:

1. Go to #8804
2. Select the 🔔 Subscribe button on the page
3. When the maintainers merge a PR that fixes the issue, our `semantic-release` bot will comment on the issue, for example: https://github.com/renovatebot/renovate/issues/23333#issuecomment-1634750025
4. When you see the `semantic-release` bot comment on #8804, you can use it right away if you're self-hosting Renovate and update to that version.
5. If you're using the Mend Renovate app you'll have to wait for the Renovate maintainers to update the version. [^1]

[^1]: https://docs.renovatebot.com/faq/#when-is-the-mend-renovate-app-updated-with-new-renovate-versions

**Reply by ghost** [reply]
thx!
